src/Libraries/DynamoUnits/Utilities.cs

```
@@ -35,8 +67,95 @@ public static double ParseExpression(string expression)
             return ForgeUnitsEngine.parseUnitless(expression);
         }
 
+        /// <summary>
+        /// Returns a formatted unit value string using unitType and unitSymbol type ids.
+        /// Sample: '12.345 m3' is a value of '12.345', of unit 'meter', of symbol 'meter cubed (m3)', of number format 'decimal', of precision '3'.
+        /// </summary>
+        /// <param name="numValue"></param>
+        /// <param name="unitTypeId"></param>
+        /// <param name="unitSymbolId"></param>
+        /// <param name="precision"></param>
+        /// <param name="decimalFormat"></param>
+        /// <returns></returns>
+        [IsVisibleInDynamoLibrary(false)]
+        public static string ReturnFormattedString(double numValue, string unitTypeId, string unitSymbolId, int precision, string numberFormat)
+        {
+            NumberFormat actualNumberFormat = numberFormat.ToString().Contains("Decimal") ? NumberFormat.Decimal : NumberFormat.Fraction;
+
+            Unit unit = Unit.ByTypeID(unitTypeId);
+            UnitSymbol unitSymbol = UnitSymbol.ByTypeID(unitSymbolId);
+
+            switch (actualNumberFormat)
+            {
+                case NumberFormat.Decimal:
+                    return UnitSymbol.StringifyDecimal(numValue, precision, unitSymbol, true);
+                case NumberFormat.Fraction:
+                    return UnitSymbol.StringifyFraction(numValue, precision, unitSymbol);
+                default:
+                    return "A number format needs to be specified.";
```

i guess we want to throw an exception here dont we?

---

Fixed.